pkg/controllers/scalablenodegroup/v1alpha1/suite_test.go

```diff
@@ -53,24 +54,75 @@ var _ = AfterSuite(func() {
 
 var _ = Describe("Examples", func() {
 	var ns *environment.Namespace
+	var desiredReplicas = ptr.Int32(5)
+	var dummyMessage = "test message"
 	var sng *v1alpha1.ScalableNodeGroup
 
 	BeforeEach(func() {
 		var err error
 		ns, err = env.NewNamespace()
 		Expect(err).NotTo(HaveOccurred())
 		sng = &v1alpha1.ScalableNodeGroup{}
+		fakeCloudProvider.NodeGroupStable = true
```

Comment:
Consider passing these directly into the cloud provider to avoid mutation. You might need to just instantiate it directly rather than use the constructor.

Reply:
I set the custom values in the last two tests -
```
fakeCloudProvider.NodeGroupStable = false
fakeCloudProvider.NodeGroupMessage = dummyMessage
...
fakeCloudProvider.WantErr = fake.RetryableError(fmt.Errorf(dummyMessage))
```

So, I just reset them before each test.

Reply:
sounds good.

Reply:
Can this be resolved?